applications/nrf_desktop/doc/hids.rst

```diff
@@ -1,6 +1,162 @@
++-----------------------------------------------+----------------------------+-------------+-----------------------------------+-----------------------------------------------+
+
+Configuration
+*************
+
+Complete the following steps to configure the module:
+
+1. Complete the basic Bluetooth configuration, as described in the Bluetooth guide.
+   During this configuration, you must enable the :option:`CONFIG_BT_PERIPHERAL` Kconfig option for every nRF Desktop peripheral.
+   When this option is enabled, the ``CONFIG_DESKTOP_HID_PERIPHERAL`` is set to ``y``, which enables the following two additional options, among others:
+
+   * :option:`CONFIG_BT_GATT_HIDS` - This is required because the HID Service module is based on the :ref:`hids_readme` implementation of the GATT Service.
+   * ``CONFIG_DESKTOP_HIDS_ENABLE`` - This enables the ``hids`` application module.
+
+   This step also enables the |GATT_HID|.
+#. Enable the :ref:`bt_conn_ctx_readme` (:option:`CONFIG_BT_CONN_CTX`).
+   This is required by the |GATT_HID|.
+#. Configure the :ref:`hids_readme`.
+   See its documentation for configuration details.
+
+    .. tip::
+      If the HID report configuration is identical to the configuration used for one of the existing devices, you can use the same |GATT_HID| configuration.
+
+The HID Service application module forwards the information about the enabled HID notifications to other application modules using ``hid_report_subscription_event``.
+These notifications are enabled by the connected Bluetooth central.
+By default, the ``hids`` application module starts forwarding the subscriptions right after the Bluetooth connection is secured.
+
+You can define additional delay for forwarding the notifications on connection (``CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY``).
+Sending the first HID report to the connected Bluetooth peer is delayed by this period of time.
+
+.. note::
+   The nRF Desktop centrals perform the GATT service discovery and reenable the HID notifications on every reconnection.
+   A HID report that is received before the subscription is reenabled will be dropped before it reaches the application.
+   The ``CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY`` is used for keyboard reference design (``nrf52_pca20037``) to make sure that the input will not be lost on reconnection with the nRF Desktop dongle.
+
+Implementation details
+**********************
+
+The HID Service application module initializes and configures the |GATT_HID|.
+The application module registers the HID report map and every HID report that was enabled in the application configuration.
+Detailed information about HID-related configuration in nRF Desktop is available in the :ref:`nrf_desktop_hid_state` documentation.
+
+Sending HID input reports
+    After subscriptions are enabled in |HID_state|, the |HID_state| sends the HID input reports as ``hid_report_event``.
```

Comment:
We need to make sure we use either HID_state of hid_state in our docs. Let's to this in the final review of docs.

Reply:
This is replaced with `HID state module` (replace command is at the bottom of the file).